common/baseinterfaces/publisher.go

```diff
@@ -0,0 +1,38 @@
+package baseinterfaces
+
+import servicebus "github.com/Azure/azure-service-bus-go"
+
+var _ BasePublisher = &PublisherSettings{}
+
+type BasePublisher interface {
```

Comment:
that will work.
just not fond of the naming. it's not go-like

what agbout `common/types.go`
and the type can just be Publisher.

so when it's used it's `common.Publisher` embedded in `queue.Publisher` or `topic.Publisher`

Reply:
agreed, that's a cleaner convention, changed